control-plane-operator/controllers/hostedcontrolplane/ignition/reconcile.go

```diff
@@ -250,25 +265,48 @@ func serializeIgnitionConfig(cfg *igntypes.Config) ([]byte, error) {
 }
 
 func setMachineConfigLabels(mc *mcfgv1.MachineConfig) {
-	mc.Labels = map[string]string{
-		"machineconfiguration.openshift.io/role": "worker",
+	if mc.Labels == nil {
+		mc.Labels = map[string]string{}
+	}
+	for k, v := range defaultMachineConfigLabels {
+		mc.Labels[k] = v
 	}
 }
 
-func reconcileIgnitionConfigMap(cm *corev1.ConfigMap, mc *mcfgv1.MachineConfig, ownerRef config.OwnerRef) error {
-	ownerRef.ApplyTo(cm)
+func reconcileICSPIgnitionConfigMap(cm *corev1.ConfigMap, icsp *v1alpha1.ImageContentSourcePolicy, ownerRef config.OwnerRef) error {
+	scheme := runtime.NewScheme()
+	v1alpha1.Install(scheme)
+	yamlSerializer := jsonserializer.NewSerializerWithOptions(
+		jsonserializer.DefaultMetaFactory, scheme, scheme,
+		jsonserializer.SerializerOptions{Yaml: true, Pretty: true, Strict: true})
+	imageContentSourceBytesBuffer := bytes.NewBuffer([]byte{})
+	err := yamlSerializer.Encode(icsp, imageContentSourceBytesBuffer)
```

Comment:
nit: error can be inlined?
Can we just yaml.marshall/unmarshall here?

Reply:
Moved the check inline.
The serializer does do some logic that is not present when just marshalling/unmarshalling. It seems it would be safer to use it.